scripts/graphql/codegen.ts

```diff
@@ -0,0 +1,156 @@
+              selection => selection.kind === Kind.FIELD && selection.name.value === "__typename",
+            );
+
+            if (!hasTypename) {
+              node.selections = [
+                { kind: Kind.FIELD, name: { kind: Kind.NAME, value: "__typename" } },
+                ...node.selections,
+              ];
+            }
+          },
+        },
+      }),
+    };
+  });
+
+const scalars = {
+  AccountNumber: "string",
+  AmountValue: "string",
+  AuditId: "string",
+  AuthorizationId: "string",
+  BIC: "string",
+  CCA2: "string",
+  CCA3: "string",
+  CardToken: "string",
+  Currency: "string",
+  Date: "string",
+  DateTime: "string",
+  EmailAddress: "string",
+  HexColorCode: "string",
+  IBAN: "string",
+  PIN: "string",
+  PhoneNumber: "string",
+  PostalCode: "string",
+  SepaCreditorIdentifier: "string",
+  SepaReference: "string",
+  TokenRequestorId: "string",
+  URL: "string",
+  Upload: "unknown",
+  WalletToken: "string",
+  join__FieldSet: "unknown",
+};
+
+const clientPlugins = [
+  "typescript",
+  "typescript-operations",
+  "typed-document-node",
+  "typescript-urql-graphcache",
+];
+
+const clientConfig = {
+  scalars,
+  dedupeOperationSuffix: true,
+  enumsAsTypes: true,
+  nonOptionalTypename: true,
+  defaultScalarType: "unknown",
+  inlineFragmentTypes: "combine",
+};
+
+const serverPlugins = ["typescript", "typescript-operations", "typescript-graphql-request"];
+
+const serverConfig = {
+  ...clientConfig,
+  federation: true,
+};
+
+const partnerSchema = file("./dist/partner-schema.gql");
+const unauthenticatedSchema = file("./dist/unauthenticated-schema.gql");
+
+const config: CodegenConfig = {
+  errorsOnly: true,
+  overwrite: true,
+
+  hooks: {
+    afterAllFileWrite: "prettier --write",
+  },
+
+  generates: {
+    [file("../../clients/onboarding/src/graphql/unauthenticated.ts")]: {
+      documents: file("../../clients/onboarding/src/graphql/unauthenticated.gql"),
+      schema: unauthenticatedSchema,
+      plugins: clientPlugins,
+      config: clientConfig,
+      documentTransforms: [{ transform: addTypenames }],
+    },
+
+    [file("../../clients/onboarding/src/graphql/introspection.json")]: {
+      schema: unauthenticatedSchema,
+      plugins: ["introspection"],
+      config: { descriptions: false },
+      hooks: {
+        afterOneFileWrite: "yarn tsx scripts/graphql/cleanIntrospectionSchema.ts",
+      },
+    },
+
+    [file("../../clients/banking/src/graphql/partner.ts")]: {
+      documents: file("../../clients/banking/src/graphql/partner.gql"),
+      schema: partnerSchema,
+      plugins: clientPlugins,
+      config: clientConfig,
+      documentTransforms: [{ transform: addTypenames }],
+    },
+
+    [file("../../clients/banking/src/graphql/unauthenticated.ts")]: {
+      documents: file("../../clients/banking/src/graphql/unauthenticated.gql"),
+      schema: unauthenticatedSchema,
+      config: clientConfig,
+      plugins: clientPlugins,
+    },
```

Comment:
it seems `documentTransforms` is missing here

Reply:
Ah yes, good catch!